src/content.js

```diff
@@ -6,32 +6,36 @@ function openRandomTab() {
   window.open(tab.getAttribute("href"));
 }
 
-let rBeingPressed = false;
-
-function onKeyDown(event) {
-  if (event.key === "r") {
-    rBeingPressed = true;
+function openRandomTabOnMiddleClick(event) {
+  if (event.which == 2) {
+    event.preventDefault();
+    openRandomTab()
   }
+}
 
-  if (rBeingPressed && event.ctrlKey && event.metaKey) {
-    openRandomTab();
-    rBeingPressed = false;
+function ignoreBrowserMiddleMouseDownEvent(event) {
+  if (event.which == 2) {
+    event.preventDefault();
   }
 }
-function onKeyUp(event) {
-  if (event.key === "r") {
-    rBeingPressed = false;
```

Comment:
I know this logic seems unnecessarily complex but I remember adding it for a reason (which I probably should've documented in the code but, alas, I did not 😅). Is the removal of this extra logic part of the "cleanup", or is it necessary to make the middle click work? If the former, then please revert the cleanup changes and only add the bare minimum to add your feature to avoid regression.

Reply:
I understand, I've been there too. This is my first time adding to an open-source project to please bare with my naiveties.

With my latest commit, I've removed any code changes that are not required to add this feature.